tests/FSharp.Compiler.ComponentTests/Conformance/Types/TypeConstraints/IWSAMsAndSRTPs/IWSAMsAndSRTPsTests.fs

```diff
@@ -1101,6 +1101,75 @@ module StaticAbstractBug =
             (Error 17, Line 9, Col 22, Line 9, Col 29, "The member 'Execute: unit -> unit' does not have the correct type to override the corresponding abstract method. Non-static member is expected.")
             (Error 783, Line 8, Col 15, Line 8, Col 25, "At least one override did not correctly implement its corresponding abstract member")
          ]
+        
+    [<FactForNETCOREAPP>]
+    let ``Produce an error when one leaves keyword "static" when implementing IWSAM in an object expression`` () =
+        Fsx """
+type IOperation =
+        static abstract member Execute: unit -> unit
+        abstract member Execute2: unit -> unit
+
+let _ =
+    { new IOperation with
+        static member Execute() = ()
+        member _.Execute2() = ()
+    }
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> typecheck
+         |> shouldFail
+         |> withDiagnostics [
+             (Error 3860, Line 8, Col 23, Line 8, Col 30, "Static members are not allowed in object expressions.")
+         ]
+         
+    [<FactForNETCOREAPP>]
+    let ``Produce an error when one leaves keyword "static" when implementing multiple IWSAM in an object expression`` () =
+        Fsx """
+type IOperation =
+        static abstract member Execute: unit -> unit
+        static abstract member Execute2: unit -> unit
+
+let _ =
+    { new IOperation with
+        static member Execute() = ()
+        static member Execute2() = ()
+    }
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> typecheck
+         |> shouldFail
+         |> withDiagnostics [
+            (Error 3860, Line 8, Col 23, Line 8, Col 30, "Static members are not allowed in object expressions.");
+            (Error 3860, Line 9, Col 23, Line 9, Col 31, "Static members are not allowed in object expressions.")
+         ]
+         
+    [<FactForNETCOREAPP>]
+    let ``Produce an error when one leaves keyword "static" when implementing static and not static members in an object expression`` () =
+        Fsx """
+type IOperation =
+        static abstract member Execute: unit -> unit
+        abstract member Execute1: unit -> unit
+        static abstract member Execute2: unit -> unit
+        abstract member Execute3: bool
+        
+let _ =
+    { new IOperation with
+        static member Execute() = ()
+        member this.Execute1() = ()
+        static member Execute2() = ()
+        member this.Execute3 = true
+    }
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> typecheck
+         |> shouldFail
+         |> withDiagnostics [
+            (Error 3860, Line 10, Col 23, Line 10, Col 30, "Static members are not allowed in object expressions.");
+            (Error 3860, Line 12, Col 23, Line 12, Col 31, "Static members are not allowed in object expressions.");
+         ]
```

Comment:
I would add one more test (unless I'm blind and it exists):
- Define an interface with one static and one instance abstract members.
- Have an object expression which implements only instance abstract.
- Make sure we can call that instance member.

No errors are expected and we can call needed member.

Reply:
Sure. It will still show the error about not implementing the static member in the object expression right? ie:

`(Error 366, Line 7, Col 5, Line 10, Col 6, "No implementation was given for 'static abstract ILogger.Execute: string -> unit'. Note that all interface members must be implemented and listed under an appropriate 'interface' declaration, e.g. 'interface ... with member ...'.")`

Reply:
```fsharp
[<Interface>]
type IAdditionOperator<'T> =
    abstract myMember: 'T * 'T -> 'T
    static member myStaticMember x = x

let a =
  {
    new IAdditionOperator<int> with
      member x.myMember (a,b) = a + b
  }

a.myMember(1,2)
```

This works today, I think for object expression, we can skip the non abstract members.

@vzarytovskii, did you mean one instance abstract member and one static plain/sealed/nonvirtual member?

Reply:
Happy to create a follow up PR about not producing diagnostic about static member in object expression.

Reply:
it finally clicks with me (object expression are freed from implementing anything static, won't generate issues), thanks @vzarytovskii and thanks @edgarfgp for nailing those internal exceptions and enhancing the messages!